src/main/java/com/learnreactivespring/handler/weather/WeatherHandler.java

```diff
@@ -0,0 +1,82 @@
+package com.learnreactivespring.handler.weather;
+
+import com.learnreactivespring.model.Weather;
+
+import com.learnreactivespring.repository.WeatherRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.server.ServerRequest;
+import org.springframework.web.reactive.function.server.ServerResponse;
+import reactor.core.publisher.Mono;
+
+@Service
+public class WeatherHandler implements IWeatherHandler {
+
+  @Autowired
+  WeatherRepository weatherRepository;
+
+
+  private WebClient webClient;
+
+  @Value("${weather.api.url}")
+  private String weatherApiURL;
+
+  @Value("${weather.api.key}")
+  private String weatherApiKey;
```

Comment:
These can go in a properties class.
Please refer to [this](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/context/properties/ConfigurationProperties.html)